fs2/src/test/scala/ReadmeConstructionCodeSnippetsTest.scala

```diff
@@ -19,95 +19,98 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import com.github.ghik.silencer.silent
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpecLike
 
-@silent final class ReadmeConstructionCodeSnippetsTest extends AnyWordSpecLike with Matchers {
+final class ReadmeConstructionCodeSnippetsTest extends AnyWordSpecLike with Matchers {
   "Cats effect Sync construction snippets should compile" in {
-    import java.util.{logging => jul}
-
-    import cats.effect.Sync
-    import cats.syntax.functor._
-    import log.effect.fs2.SyncLogWriter._
-    import log.effect.{LogLevels, LogWriter}
-    import org.{log4s => l4s}
-
-    sealed abstract class App[F[_]](implicit F: Sync[F]) {
-      val log4s1: F[LogWriter[F]] =
-        F.delay(l4s.getLogger("test")) map log4sLog[F]
-
-      val log4s2: F[LogWriter[F]] = log4sLog("a logger")
-
-      val log4s3: F[LogWriter[F]] = {
-        case class LoggerClass()
-        log4sLog(classOf[LoggerClass])
-      }
-
-      val jul1: F[LogWriter[F]] =
-        F.delay(jul.Logger.getLogger("a logger")) map julLog[F]
-
-      val jul2: F[LogWriter[F]] = julLog
-
-      val scribe1: F[LogWriter[F]] =
-        F.delay(scribe.Logger("a logger")) map scribeLog[F]
-
-      val scribe2: F[LogWriter[F]] = scribeLog("a logger")
-
-      val scribe3: F[LogWriter[F]] = {
-        case class LoggerClass()
-        scribeLog(classOf[LoggerClass])
-      }
-
-      val console1: LogWriter[F] = consoleLog
-
-      val console2: LogWriter[F] = consoleLogUpToLevel(LogLevels.Warn)
-
-      val noOp: LogWriter[F] = noOpLog[F]
-    }
+    """
+    |import java.util.{logging => jul}
+
+    |import cats.effect.Sync
+    |import cats.syntax.functor._
+    |import log.effect.fs2.SyncLogWriter._
+    |import log.effect.{LogLevels, LogWriter}
+    |import org.{log4s => l4s}
+    |
+    |sealed abstract class App[F[_]](implicit F: Sync[F]) {
+    |  val log4s1: F[LogWriter[F]] =
+    |    F.delay(l4s.getLogger("test")) map log4sLog[F]
+    |
+    |  val log4s2: F[LogWriter[F]] = log4sLog("a logger")
+    |
+    |  val log4s3: F[LogWriter[F]] = {
+    |    case class LoggerClass()
+    |    log4sLog(classOf[LoggerClass])
+    |  }
+    |
+    |  val jul1: F[LogWriter[F]] =
+    |    F.delay(jul.Logger.getLogger("a logger")) map julLog[F]
+    |
+    |  val jul2: F[LogWriter[F]] = julLog
+    |
+    |  val scribe1: F[LogWriter[F]] =
+    |    F.delay(scribe.Logger("a logger")) map scribeLog[F]
+    |
+    |  val scribe2: F[LogWriter[F]] = scribeLog("a logger")
+    |
+    |  val scribe3: F[LogWriter[F]] = {
+    |    case class LoggerClass()
+    |    scribeLog(classOf[LoggerClass])
+    |  }
+    |
+    |  val console1: LogWriter[F] = consoleLog
+    |
+    |  val console2: LogWriter[F] = consoleLogUpToLevel(LogLevels.Warn)
+    |
+    |  val noOp: LogWriter[F] = noOpLog[F]
+    |}
+    |""".stripMargin should compile
```

Comment:
This I think makes sense to test for compilation this way but, anyway, the nested sealed class is problematic again in Scala 3:
```
[error] -- Error: /Users/jsirocchi/code/log-effect/fs2/src/test/scala/ReadmeConstructionCodeSnippetsTest.scala:27:63
[error] 27 |@nowarn final class ReadmeConstructionCodeSnippetsTest extends AnyWordSpecLike with Matchers {
[error]    |                                                               ^
[error]    | bad parameter reference App.this.F at inlining
[error]    | the parameter is type F in class App but the prefix (App.this : App[F])
[error]    | does not define any corresponding arguments.
[error]    | idx = 0, args = List()
```
Looks related to https://github.com/lampepfl/dotty/issues/12508 but may be another instance. 3.1.2-RC2 still has this issue